platform/android/src/style/layers/layer.cpp

```diff
@@ -144,6 +144,25 @@ namespace android {
         }
     }
 
+    jni::String Layer::getSourceLayer(jni::JNIEnv& env) {
```

Comment:
I set a bad example here (my apologies), but a more extensible way of doing this would be something like:

```
struct Evaluator {
    std::string operator()(mbgl::style::BackgroundLayer&) { return {}; }
    std::string operator()(mbgl::style::RasterLayer&) { return {}; }
    std::string operator()(mbgl::style::CustomLayer&) { return {}; }

    template <class VectorLayer>
    std::string operator()(VectorLayer& layer) {
        return layer.getSourceLayer();
    }
};

...

        std::string sourceLayerId = layer.accept(Evaluator());
```

maybe it's better though to clean this up separately in a pr and do the others in this class at the same time (might be quite some overlap).